ui/components/multichain/asset-picker-amount/asset-picker-modal/asset-picker-modal.tsx

```diff
@@ -0,0 +1,256 @@
+import React, { useState } from 'react';
+import { useSelector, useDispatch } from 'react-redux';
+import { useHistory } from 'react-router-dom';
+import classnames from 'classnames';
+import { isEqual } from 'lodash';
+import {
+  Modal,
+  ModalContent,
+  ModalOverlay,
+  ModalHeader,
+  TextFieldSearch,
+  Box,
+} from '../../../component-library';
+import {
+  BlockSize,
+  Size,
+  BorderRadius,
+  BackgroundColor,
+} from '../../../../helpers/constants/design-system';
+import { useI18nContext } from '../../../../hooks/useI18nContext';
+import {
+  getNativeCurrencyImage,
+  getSelectedAccountCachedBalance,
+  getSelectedAddress,
+  getShouldHideZeroBalanceTokens,
+} from '../../../../selectors';
+import { SEND_ROUTE } from '../../../../helpers/constants/routes';
+
+import { PRIMARY, SECONDARY } from '../../../../helpers/constants/common';
+import {
+  getNativeCurrency,
+  getTokens,
+} from '../../../../ducks/metamask/metamask';
+import {
+  AssetType,
+  TokenStandard,
+} from '../../../../../shared/constants/transaction';
+import { useTokenTracker } from '../../../../hooks/useTokenTracker';
+import { updateSendAsset, Asset } from '../../../../ducks/send';
+import { useUserPreferencedCurrency } from '../../../../hooks/useUserPreferencedCurrency';
+import { useCurrencyDisplay } from '../../../../hooks/useCurrencyDisplay';
+import TokenCell from '../../../app/token-cell';
+import { TokenListItem } from '../../token-list-item';
+
+type AssetPickerModalProps = {
+  isOpen: boolean;
+  onClose: () => void;
+  asset: Asset;
+};
```

Comment:
A comment for each prop would be helpful; I'm not sure what `asset` represents.

Reply:
done :)